File: src/create/prototype/PersonClone.java
package create.prototype;

/**
 * Created by Sun on 4/2/2017.
 */

public class PersonClone implements Cloneable {

    private String name;

    public PersonClone(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    @Override
    public PersonClone clone() throws CloneNotSupportedException {
        return (PersonClone) super.clone();
    }
}

File: src/create/prototype/PersonClone.kt
package create.prototype

/**
 * Created by Sun on 4/2/2017.
 */

class PersonClone(val name: String) : Cloneable {

    @Throws(CloneNotSupportedException::class)
    public override fun clone(): PersonClone {
        return super.clone() as PersonClone
    }
}
